src/compiler/config/validate-paths.ts

```diff
@@ -56,7 +73,7 @@ export const validatePaths = (config: d.ValidatedConfig) => {
     }
   }
```

Comment:
I didn't call it out super well in [this comment,](https://github.com/ionic-team/stencil/pull/4329#discussion_r1181814212) but is this mutation still necessary? Should we move this property into `ConfigPaths`?

Reply:
oh yeah I think we should change that too while we're in here - I make this change:

```diff
diff --git a/src/compiler/config/validate-paths.ts b/src/compiler/config/validate-paths.ts
index 910ff5050..4208d88d5 100644
--- a/src/compiler/config/validate-paths.ts
+++ b/src/compiler/config/validate-paths.ts
@@ -65,11 +65,11 @@ export const validatePaths = (config: d.Config): ConfigPaths => {
   }
 
   if (config.writeLog) {
-    if (typeof config.buildLogFilePath !== 'string') {
-      config.buildLogFilePath = DEFAULT_BUILD_LOG_FILE_NAME;
-    }
-    if (!isAbsolute(config.buildLogFilePath)) {
-      config.buildLogFilePath = join(config.rootDir, config.buildLogFilePath);
+    validatedPaths.buildLogFilePath =
+      typeof config.buildLogFilePath === 'string' ? config.buildLogFilePath : DEFAULT_BUILD_LOG_FILE_NAME;
+
+    if (!isAbsolute(validatedPaths.buildLogFilePath)) {
+      validatedPaths.buildLogFilePath = join(rootDir, config.buildLogFilePath);
     }
   }
```